Fitbit.Portable/JsonDotNetSerializer.cs

```diff
@@ -23,6 +23,7 @@ internal T Deserialize<T>(string data)
         {
             if (string.IsNullOrWhiteSpace(data))
             {
+                //TO DO: Is this the behavior we want? Or should we be throwing an exception?
```

Comment:
Personally I would say yes if it's empty string data then it should return default of the type which in most cases will be null. I would prefer to check if something is null and then throw the correct exception or deal with it on a case by case than have to wrap every deserialization call in a try/catch each time as it would look messy and makes me think it's a code smell.